src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs

```diff
@@ -67,7 +84,7 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
             {
                 bindingContext.ModelState.TryAddModelError(
                     bindingContext.ModelName,
-                    exception,
+                    exception.InnerException ?? exception,
```

Comment:
TryAddModelError rewrites the message for FormatExcpetions and OverflowException TypeConverter.ConvertFrom throws System exception but the inner exception is a FormatException.